File: packages/python/qsu/math/max.py
def max(*args):
	val = args[0] if len(args) > 0 and isinstance(args[0], (list, tuple)) else args
	result = None

	for i in range(len(val)):
		value = val[i]

		# `bool` is a subclass of `int`, so it has to be rejected explicitly, as `sum` does.
		if not isinstance(value, (int, float)) or isinstance(value, bool):
			continue

		# `nan` loses every comparison, so it would win by being seen first and then never
		# being replaced.
		if value != value:
			continue

		if result is None or value > result:
			result = value

	return result

File: packages/javascript/lib/math/max.ts
export function max(...args: any[]): number | null;

export function max(...args: Array<number>): number | null;

export function max(...args: Array<number> | number[]): number | null {
	const val = args.length > 0 && Array.isArray(args[0]) ? args[0] : args;
	let result: number | null = null;

	for (let i = 0, iLen = val.length; i < iLen; i += 1) {
		const value = val[i];

		// `NaN` loses every comparison, so it would win by being seen first and then never
		// being replaced. Anything that is not a number is skipped, as `sum` does.
		if (typeof value === 'number' && !Number.isNaN(value) && (result === null || value > result)) {
			result = value;
		}
	}

	return result;
}
